code/game/machinery/_machines_base/machinery_components.dm

```diff
@@ -29,6 +29,10 @@ GLOBAL_LIST_INIT(machine_path_to_circuit_type, cache_circuits_by_build_path())
 				LAZYINITLIST(uncreated_component_parts)
 				for(var/type in req_components)
 					uncreated_component_parts[type] += (req_components[type] || 1)
+		if(initial_access && length(initial_access) > 0)
+			// Auto-create a network lock.
+			var/obj/item/stock_parts/network_lock/lock = install_component(/obj/item/stock_parts/network_lock, refresh_parts = FALSE)
+			lock.grants = initial_access
```

Comment:
I don't get it; this just won't work as far as I can tell. This is supposed to be a list of file names, but you're potentially feeding in a list of lists.

Reply:
I'm never supposed to be feeding it in a list of lists. List of lists is invalid format.